app/src/main/java/org/schabi/newpipe/fragments/list/channel/ChannelTabFragment.java

```diff
@@ -114,6 +117,27 @@ public void setTitle(final String title) {
     public void handleResult(@NonNull final ChannelTabInfo result) {
         super.handleResult(result);
 
+        // FIXME this is a really hacky workaround, to avoid storing useless data in the fragment
+        //  state. The problem is, `ReadyChannelTabListLinkHandler` might contain raw JSON data that
+        //  uses a lot of memory (e.g. ~800KB for YouTube). While 800KB doesn't seem much, if
+        //  you combine just a couple of channel tab fragments you easily go over the 1MB
+        //  save&restore transaction limit, and get `TransactionTooLargeException`s. A proper
+        //  solution would require rethinking about `ReadyChannelTabListLinkHandler`s.
+        if (tabHandler instanceof ReadyChannelTabListLinkHandler) {
+            try {
+                // once `handleResult` is called, the parsed data was already saved to cache, so
+                // we can discard any raw data in ReadyChannelTabListLinkHandler and create a
+                // link handler with identical properties, but without any raw data
+                tabHandler = result.getService()
+                        .getChannelTabLHFactory()
+                        .fromQuery(tabHandler.getId(), tabHandler.getContentFilters(),
+                                tabHandler.getSortFilter());
```

Comment:
This causes a crash. Use media.ccc.de service, open a channel / conference. `getChannelTabLHFactory()` is `null` for that service:

https://github.com/TeamNewPipe/NewPipeExtractor/blob/f276caf54a85a271904943e4aa6c57efd4249fec/extractor/src/main/java/org/schabi/newpipe/extractor/services/media_ccc/MediaCCCService.java#L53-L56


## Exception
* __User Action:__ ui error
* __Request:__ ACRA report
* __Content Country:__ DE
* __Content Language:__ de-
* __App Language:__ en
* __Service:__ none
* __Version:__ 0.26.0
* __OS:__ Linux Android 12 - 32
<details><summary><b>Crash log </b></summary><p>

```
java.lang.NullPointerException: Attempt to invoke virtual method 'org.schabi.newpipe.extractor.linkhandler.ListLinkHandler org.schabi.newpipe.extractor.linkhandler.ListLinkHandlerFactory.fromQuery(java.lang.String, java.util.List, java.lang.String)' on a null object reference
	at org.schabi.newpipe.fragments.list.channel.ChannelTabFragment.handleResult(ChannelTabFragment.java:133)
	at org.schabi.newpipe.fragments.list.channel.ChannelTabFragment.handleResult(ChannelTabFragment.java:37)
	at org.schabi.newpipe.fragments.list.BaseListInfoFragment.lambda$startLoading$0(BaseListInfoFragment.java:150)
	at org.schabi.newpipe.fragments.list.BaseListInfoFragment.$r8$lambda$yX6S2nKGSbR70YowNdSpGemC7q4(BaseListInfoFragment.java:0)
	at org.schabi.newpipe.fragments.list.BaseListInfoFragment$$ExternalSyntheticLambda4.accept(R8$$SyntheticClass:0)
	at io.reactivex.rxjava3.internal.observers.ConsumerSingleObserver.onSuccess(ConsumerSingleObserver.java:62)
	at io.reactivex.rxjava3.internal.operators.single.SingleObserveOn$ObserveOnSingleObserver.run(SingleObserveOn.java:81)
	at io.reactivex.rxjava3.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:123)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7870)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)

```
</details>
<hr>